src/shared/vscode/window.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Components associated with {@link vscode.window}.
+ */
+export interface Window {
+    statusBar: StatusBar
+    inputBox: InputBox
+    message: Message
+    progress: Progress
+    dialog: Dialog
+}
+
+/**
+ * Actions associated with status bars in {@link vscode.window}.
+ */
+export interface StatusBar {
+    /**
+     * See {@link vscode.window.setStatusBarMessage}.
+     */
+    setMessage(message: string, hideAfterTimeout?: number): vscode.Disposable
+}
+
+/**
+ * Actions associated with input boxes in {@link vscode.window}.
+ */
+export interface InputBox {
+    /**
+     * See {@link vscode.window.showInputBox}.
+     */
+    show(options?: vscode.InputBoxOptions, token?: vscode.CancellationToken): Thenable<string | undefined>
+}
+
+/**
+ * Actions associated with showing messages in {@link vscode.window}.
+ */
+export interface Message {
+    /**
+     * See {@link vscode.window.showErrorMessage}.
+     */
+    showError(message: string, ...items: string[]): Thenable<string | undefined>
```

Comment:
You might as well add showInformation and showWarning from the start.

Reply:
Yeah, I was just adding the stuff I needed for the S3 integration. Will add those two as well.